src/tracker-component.js

```diff
@@ -79,11 +74,31 @@ const track = <T>(config : Config, trackingType : TrackingType, trackingData : T
 const trackCartEvent = <T>(config : Config, cartEventType : CartEventType, trackingData : T) : Promise<void> =>
     track(config, 'cartEvent', { ...trackingData, cartEventType });
 
+const allCarts = JSON.parse(localStorage.getItem('paypal-cr-cart') || '{}');
+
 export const Tracker = (config : Config) => ({
     view:           (data : { pageUrl : string }) => track(config, 'view', data),
-    addToCart:      (data : Cart) => trackCartEvent(config, 'addToCart', data),
+    addToCart:      (data : Cart) => {
+        const cart = (allCarts[data.cartId] || { 'items': [], 'price': 0 });
+        data.items = (data.items || []).concat(cart.items);
+        data.price += (cart.price || 0);
+        allCarts[(data.cartId || 0)] = data;
+
+        localStorage.setItem('paypal-cr-cart', JSON.stringify(data));
+        trackCartEvent(config, 'setCart', data);
+    },
     setCart:        (data : Cart) => trackCartEvent(config, 'setCart', data),
-    removeFromCart: (data : RemoveCart) => trackCartEvent(config, 'removeFromCart', data),
+    removeFromCart: (data : Cart) => {
+        const cart = allCarts[(data.cartId)];
+        data.items = (data.items || []).filter(item => !(cart.items).includes(item));
```

Comment:
`removeFromCart` won't work because `Array.prototype.includes` uses shallow equality checking.
For example, if `cart` is `{ 'a': { items: [{ id: 'b' }] } }` and `cart.items` is `{ cartId: 'a', items: [{ id: 'b' }] }`, it won't remove the item with id `'b'` because `{ id: 'b' }` and `{ id: 'b' }` are not shallow equal.